supervision/dataset/formats/yolo.py

```diff
@@ -167,10 +167,19 @@ def load_yolo_annotations(
             annotations[image_path] = Detections.empty()
             continue
 
-        image = cv2.imread(image_path)
+        # PIL is much faster than cv2 for checking image shape and mode: https://github.com/roboflow/supervision/issues/1554
+        image = Image.open(image_path)
         lines = read_txt_file(file_path=annotation_path, skip_empty=True)
-        h, w, _ = image.shape
+        w, h = image.size
         resolution_wh = (w, h)
+        if image.mode != "RGB":
```

Comment:
@patel-zeel looks like we can simplify the code here. There is no need for nested ifs.
```python
if image.mode not in ("RGB", "L"):
    raise ValueError(
        f"Images must be 'RGB' or 'grayscale', but {image_path} mode is '{image.mode}'."
    )
```